provider/server/server.go

```diff
@@ -93,6 +94,7 @@ func (s *Server) loggerFrom(ctx context.Context, keysWithValues ...interface{})
 type MachineClassRegistry interface {
 	Get(volumeClassName string) (*iri.MachineClass, bool)
 	List() []*iri.MachineClass
+	GetResources(context.Context) (*mcr.Host, error)
```

Comment:
Please keep the interface simple until there is a need for this change. 
The  `GetResources` from `(m *Mcr)` does not have any important reference to `Mcr` which is not known outside of the function. 
```
func (m *Mcr) GetResources(ctx context.Context) (*Host, error) {
	return GetResources(ctx, m.disableHugepages)
}
```